### PR TITLE
Add messaging session

### PR DESCRIPTION
**Pull request by ltrung**
**Description of changes:**
Adding new APIs to create a web socket connection to receive chime chat message. 
**Testing**

1. Have you successfully run `npm run build:release` locally? Yes
2. How did you test these changes? 
- Unit test
- Manually test using the new demo app
3. Can these changes be tested using the demo application? If yes, which demo application can be used to test it? 
Yes, the changes can be tested using the new demo app found under demos/browser/app/messagingSession.
4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved? Yes
5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved? Yes

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,6 +130,28 @@ These objects contain all the information needed for a client application using
...
+import * as Chime from 'aws-sdk/clients/chime';
+
+import {
+  ConsoleLogger,
+  DefaultMessagingSession,
+  LogLevel,
+  MessagingSessionConfiguration,
+} from 'amazon-chime-sdk-js';
+
+this.logger = new ConsoleLogger('SDK', LogLevel.INFO);
```
Comment (simmkyu) [inline README.md:146]: - Recommend having local variables over `this`: `this.logger =` => `const logger =` (as in your previous revision)
- `SDK` => `MyLogger`

File: README.md [hunk 2]
```diff
@@ -129,6 +130,28 @@ These objects contain all the information needed for a client application using
...
+const chime = new Chime({ region: 'us-east-1' });
+const endpoint = await chime.getMessagingSessionEndpoint().promise();
+
+this.configuration = new MessagingSessionConfiguration(this.userArn, this.sessionId, endpoint.Endpoint.Url, chime, AWS);
```
Comment (simmkyu) [inline README.md:150]: Explain userArn and sessionId 

```
const userArn = /* ... explain ... */;
const sessionId = /* ... explain ... */;
```

File: README.md [hunk 3]
```diff
@@ -129,6 +130,28 @@ These objects contain all the information needed for a client application using
...
+this.session = new DefaultMessagingSession(this.configuration, this.logger);
```
Comment (simmkyu) [inline README.md:151]: `const messagingSession` to match the usage examples below

File: README.md [hunk 4]
```diff
@@ -129,6 +130,28 @@ These objects contain all the information needed for a client application using
...
+this.session.addObserver(this);
```
Comment (simmkyu) [inline README.md:152]: Remove the following two lines, or replace `addObserver(this)`  with an actual example.

**Review by simmkyu — COMMENTED** [review]
Left comments on the README

File: README.md (hunk 4 above)
Comment (ltrung) [reply README.md]: Fixed.

File: README.md (hunk 3 above)
Comment (ltrung) [reply README.md]: Fixed.

File: README.md [hunk 5]
```diff
@@ -129,6 +130,28 @@ These objects contain all the information needed for a client application using
...
+  DefaultMessagingSession,
+  LogLevel,
+  MessagingSessionConfiguration,
+} from 'amazon-chime-sdk-js';
+
+const logger = new ConsoleLogger('SDK', LogLevel.INFO);
+const chime = new Chime({ region: 'us-east-1' });
+const endpoint = await chime.getMessagingSessionEndpoint().promise();
+
+const userArn = /* The userArn */;
```
Comment (simmkyu) [inline README.md:150]: Please have the follow-up items to improve `userArn` and `sessionId` comments

**Review by simmkyu — CHANGES_REQUESTED** [review]
Please remove the `patch.diff` file

**Review by simmkyu — APPROVED** [review]

**Comment by john-raymon** [comment]
@ltrung Hey, I'm currently trying to pick up/learn how to implement the aws chime sdk in a Node.js/Vue.js mono web app, by using the `demos/browser` demo. 

I'm confused by the part where secret keys and other secure credentials are returned to the client, is this something done solely for the demo/example?
https://github.com/aws/amazon-chime-sdk-js/blob/239a41d840d2df02a6f25bfcbb844b87e0a11bc5/demos/browser/server.js#L93

**Comment by michhyun1** [comment]
> @ltrung Hey, I'm currently trying to pick up/learn how to implement the aws chime sdk in a Node.js/Vue.js mono web app, by using the `demos/browser` demo.
> 
> I'm confused by the part where secret keys and other secure credentials are returned to the client, is this something done solely for the demo/example?
> https://github.com/aws/amazon-chime-sdk-js/blob/239a41d840d2df02a6f25bfcbb844b87e0a11bc5/demos/browser/server.js#L93

Certain aws credentials are needed to make these requests, check https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html

**Pull request merged**